src/main/java/core/basesyntax/strategy/handler/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.strategy.handler;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    private final FruitDao fruitDao;
+
+    public PurchaseOperationHandler(FruitDao fruitDao) {
+        this.fruitDao = fruitDao;
+    }
+
+    @Override
+    public void process(FruitTransaction transaction) {
+        Integer currentQuantity = fruitDao.get(transaction.getFruit());
+        int purchaseQuantity = transaction.getQuantity();
+        if (purchaseQuantity <= 0 && purchaseQuantity > currentQuantity) {
+            throw new IllegalArgumentException("Invalid purchase quantity - " + purchaseQuantity);
+        }
+        fruitDao.getStorage().put(transaction.getFruit(), currentQuantity - purchaseQuantity);
```

Comment:
Before updating the storage with the new quantity, you should check if the result will not be negative. If `currentQuantity - purchaseQuantity` results in a negative number, you should throw a `RuntimeException` to indicate that there is not enough stock for the purchase.